tests/test_infer.py

```diff
@@ -20,27 +23,124 @@ def check_uniform_type(in_list):
     )
 
 
-def test_upperlimit(tmpdir, hypotest_args):
+def test_toms748_scan(tmpdir, hypotest_args):
+    """
+    Test the upper limit toms748 scan returns the correct structure and values
+    """
+    _, data, model = hypotest_args
+    results = pyhf.infer.intervals.upper_limits.toms748_scan(
+        data, model, 0, 5, rtol=1e-8
+    )
+    assert len(results) == 2
+    observed_limit, expected_limits = results
+    observed_cls = pyhf.infer.hypotest(
+        observed_limit,
+        data,
+        model,
+        model.config.suggested_init(),
+        model.config.suggested_bounds(),
+    )
+    expected_cls = np.array(
+        [
+            pyhf.infer.hypotest(
+                expected_limits[i],
+                data,
+                model,
+                model.config.suggested_init(),
+                model.config.suggested_bounds(),
+                return_expected_set=True,
+            )[1][i]
+            for i in range(5)
+        ]
+    )
+    assert observed_cls == pytest.approx(0.05)
+    assert expected_cls == pytest.approx(0.05)
+
+
+def test_toms748_scan_rtol_warning(hypotest_args):
+    """
+    Test the UserWarning is raised if no rtol is given.
+    """
+    _, data, model = hypotest_args
+    with pytest.raises(UserWarning):
+        pyhf.infer.intervals.upper_limits.toms748_scan(data, model, 0, 5)
```

Comment:
This will require an update to 

https://github.com/scikit-hep/pyhf/blob/f7e974f7c65524e5240c871154c00ba8e954f0bd/.github/workflows/lower-bound-requirements.yml#L35-L42

given that the current workflow ignores all warnings, and so this fails for the Minimum supported dependencies workflow. Do this in another PR.

```console
$ PYTHONWARNINGS='default' pytest tests/test_infer.py
```

works, but I don't think that will work for the entire test suite.

Reply:
Just get rid of the UserWarning and set a default. Henry has warned us before that `UserWarnings` might as well be useless.